addons/SimpleTODO/TODOItem.gd

```diff
@@ -4,18 +4,184 @@ extends HBoxContainer
 onready var text_field = $Text
 onready var label_hack = $Text/HackLabel
 
-signal delete
+var item_placement_holder: Panel
+var main: Control
+var undo_redo: UndoRedo
+
+var initial_item_index = 0
+
+# The main TODO scene column container node
+var main_column_container: HBoxContainer
+
+var parent_column: PanelContainer
+
+var next_parent_column: PanelContainer
+
+var current_drag_item_index = 0
+
+var is_dragging = false
+
+var item_margin = 20
 
 func _ready() -> void:
+	undo_redo = main.undo_redo
+	item_placement_holder = main.item_placement_holder
+	next_parent_column = parent_column
+
 	get_node("Button").icon = get_icon("Remove", "EditorIcons")
 	call_deferred("text_changed")
 
 func delete_pressed() -> void:
-	emit_signal("delete")
+	delete_item()
 
 func request_save() -> void:
 	get_tree().get_nodes_in_group("__todo_plugin__").front().save_data()
 
 func text_changed() -> void:
 	label_hack.text = text_field.text
 	text_field.rect_min_size.y = label_hack.get_minimum_size().y + 8
+
+func _process(_delta):
+	if is_dragging:
+		var mouse_position = main.get_local_mouse_position()
+		
+		var item_under_mouse = get_column_item_from_mouse_position()
+		if item_under_mouse:
+			var column = item_under_mouse.column
+			var column_items = column.item_container
+			var item = item_under_mouse.item
+			next_parent_column = column
+			
+			if not column_items == item_placement_holder.get_parent():
+				item_placement_holder.get_parent().remove_child(item_placement_holder)
+				column_items.add_child(item_placement_holder)
+			
+			if item:
+				var item_index = item.get_index()
+				item_placement_holder.get_parent().move_child(item_placement_holder, item_index)
+				current_drag_item_index = item_index
+			elif column_items.get_child_count() <= 0:
+				item_placement_holder.get_parent().move_child(item_placement_holder, 0)
+				current_drag_item_index = 0
+		
+		rect_position = mouse_position
+
+func get_column_from_mouse_position() -> PanelContainer:
+	var mouse_position = main.column_container.get_local_mouse_position()
+	for i in main.column_container.get_child_count():
+		var child = main.column_container.get_child(i)
+		var rect: Rect2 = child.get_rect()
+		if rect.has_point(mouse_position):
+			return child
+	return null
```

Comment:
Duplicated code. You could call this from the column, as you already have a reference.

Reply:
Will check on this and probably fix it. I think the column and item codes are different though I haven't looked at it in a bit.